packages/lazy-import/lib/index.js

```diff
@@ -0,0 +1,114 @@
+/**
+ * External dependencies
+ */
+const npa = require( 'npm-package-arg' );
+const semver = require( 'semver' );
+const execa = require( 'execa' );
+const { join } = require( 'path' );
+const { createHash } = require( 'crypto' );
+
+/**
+ * @typedef WPLazyImportOptions
+ *
+ * @property {()=>void} onInstall Callback to invoke when install starts.
+ */
+
+/**
+ * Returns an md5 hash of the given string.
+ *
+ * @param {string} text Text for which to generate hash.
+ *
+ * @return {string} md5 hash of string.
+ */
+function md5( text ) {
+	return createHash( 'md5' )
+		.update( text )
+		.digest( 'hex' );
+}
+
+/**
+ * Given an arg string as would be passed to `npm install`, returns a string to
+ * use as the alias name of the package to install locally if needed.
+ *
+ * @param {string} arg A string that you might pass to `npm install`.
+ *
+ * @return {string} Module to use as local package alias.
+ */
+function getLocalModuleName( arg ) {
+	return '@wordpress/lazy-import.' + md5( arg );
+}
+
+/**
+ * Installs npm package by arg name at the given prefix path. Creates the prefix
+ * path if it doesn't yet exist.
+ *
+ * @param {string} arg   Package install arg (name and version specification).
+ * @param {string} alias Alias to use for local module name.
+ *
+ * @return {Promise<void>} Promise resolving once package is installed.
+ */
+async function install( arg, alias ) {
+	await execa( 'npm', [ 'install', '--no-save', alias + '@npm:' + arg ] );
+}
+
+/**
+ * Given an arg string as would be passed to `npm install`, requires the package
+ * corresponding to the name and specifier parsed from the arg. If the package
+ * is not installed (or not installed at that version), the package is installed
+ * to a temporary directory at `node_modules/.wp-lazy` relative to where the
+ * current working directory.
+ *
+ * @param {string}                       arg       A string that you might pass
+ *                                                 to `npm install`.
+ * @param {Partial<WPLazyImportOptions>} [options] Optional options object.
+ *
+ * @return {Promise<NodeRequire>} Promise resolving to required module.
+ */
+async function lazyImport( arg, options = {} ) {
+	const { rawSpec, name } = npa( arg );
```

Comment:
I see they use an abbreviation in their docs but when you see it for the first time you are completely lost 😃

```suggestion
	const { rawSpec, name } = npmPackageArg( arg );
```

Reply:
> I see they use an abbreviation in their docs but when you see it for the first time you are completely lost 😃

I agree 👍  Will change it.

Reply:
> > I see they use an abbreviation in their docs but when you see it for the first time you are completely lost 😃
> 
> I agree 👍 Will change it.

Updated to avoid abbreviation in 24e0e45efd.